File: seeker/snippet/product.go
//date: 2024-12-20T16:55:54Z
//url: https://api.github.com/gists/ca4bbc67742e9adca587643ac1ca2b5f
//owner: https://api.github.com/users/Muhafiz22

func (p *Product) Get(Products []Product) { //to Get Product details
	//It Accepts Products slice to check if the product to be added is already in the inventory's product list

	fmt.Print("Enter Product ID:")
	if _, err := fmt.Scan(&p.ID); err != nil {
		fmt.Print(p.ID)
		fmt.Println(" Invalid input, Please Enter valid ID")
	}

	existingProduct := FindProductById(Products, p.ID) // to find if same Product is already exixting in inventory or not

	if existingProduct != nil { // if same Product is already existing in inventory then add in existing Product details

		fmt.Print("Enter additional Quantity:")
		var QdditionalQuantity int
		fmt.Scan(&QdditionalQuantity)

		existingProduct.Quantity += QdditionalQuantity
		p.ID = existingProduct.ID
		p.Name = existingProduct.Name
		p.Price = existingProduct.Price
		p.Quantity = existingProduct.Quantity
	} else { // adding new Product to inventory

		fmt.Println("Enter Product Name:")
		fmt.Scan(&p.Name)
		fmt.Println("Enter Product Price:")
		fmt.Scan(&p.Price)
		fmt.Println("Enter Product Quantity:")
		fmt.Scan(&p.Quantity)
	}
}

File: seeker/snippet/Product.java
//date: 2025-05-27T16:48:45Z
//url: https://api.github.com/gists/72ffc3e06a5c416ead357910b5decd99
//owner: https://api.github.com/users/rdurelli

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private double price;

    // Construtores, Getters e Setters
    public Product() {}

    public Product(String name, double price) {
        this.name = name;
        this.price = price;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ProductRepository extends JpaRepository<Product, Long> {
}

